.github/workflows/ci.yml

```diff
@@ -65,21 +65,46 @@ jobs:
         coverage-driver:
           - "pcov"
           - "xdebug"
+          - "xdebug3"
 
         dependencies:
           - "lowest"
           - "highest"
 
+        exclude:
+          - php-version: "8.0"
+            coverage-driver: "xdebug"
+            dependencies: "lowest"
+
+          - php-version: "8.0"
+            coverage-driver: "xdebug"
+            dependencies: "highest"
+
     steps:
       - name: "Checkout"
         uses: "actions/checkout@v2"
 
-      - name: "Install PHP with extensions"
-        uses: "shivammathur/setup-php@v1"
+      - name: "Install PHP with PCOV or Xdebug 2"
+        if: "matrix.coverage-driver != 'xdebug3'"
+        uses: "shivammathur/setup-php@v2"
         with:
           php-version: "${{ matrix.php-version }}"
           coverage: "${{ matrix.coverage-driver }}"
 
+      - name: "Install PHP with Xdebug 3"
+        if: "matrix.coverage-driver == 'xdebug3'"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          php-version: "${{ matrix.php-version }}"
+          coverage: "none"
```

Comment:
@shivammathur

Can you help us out here?

With the current configuration, Xdebug 3 will not be installed, but neither will the PCOV module be enabled.

When we apply this patch, Xdebug 3 will be installed, but the PCOV module will be enabled. In this case PCOV will be selected as code coverage driver. 

```suggestion
```

Reply:
@localheinz checking this.

Reply:
@localheinz I released [`2.3.1`](https://github.com/shivammathur/setup-php/releases/tag/2.3.1).
`coverage: xdebug` for PHP 8 will now correctly set driver to `Xdebug` as it does for other versions. No need to specify in extensions.

Reply:
@shivammathur

Nice!

Is it possible to explicitly enable Xdebug 3 instead of Xdebug 3?

The aim is to run tests with Pcov, Xdebug 2, and Xdebug 3 for

- PHP 7.3
- PHP 7.4

and to run tests with Pcov and XDebug 3 for

- PHP 8.0

That is, we need to be able to explicitly set the version of Xdebug.

What do you think?

Reply:
@localheinz 
Xdebug3 is currently setup only for PHP 8.0. In next release I can add a `coverage: xdebug3` option for PHP 7.2 and above. I was waiting for alpha tag for Xdebug 3.0 to do this.

Reply:
@localheinz 
Updated the release. Now you can setup both Xdebug 2 and 3 for PHP 7.2, 7.3 and 7.4.

*Xdebug 2 on 7.3*
```yml
php-version: 7.3
coverage: xdebug
```
*Xdebug 3 on 7.3*
```yml
php-version: 7.3
coverage: xdebug3
```

So the workflow may look like this, excluding `xdebug` on PHP 8 as it will always will setup Xdebug 3 for both `coverage: xdebug` and `coverage: xdebug3`.
```yml
...
strategy:
  matrix:
    php-version: ['7.2', '7.3', '7.4', '8.0']
    coverage: [xdebug, xdebug3, pcov, none]
    dependencies: [lowest, highest]
    exclude:
      - php-version: '8.0'
         coverage: xdebug
         dependencies: lowest
      - php-version: '8.0'
         coverage: xdebug
         dependencies: highest
steps:
....
  - uses: shivammathur/setup-php@v2
     with:
        php-version: ${{ matrix.php-version }}
        coverage: ${{ matrix.coverage }}
....
```

When coverage is `pcov` or `none`, It will remove `Xdebug` or you can add those in excludes.

Reply:
Thank you, @shivammathur!